File: Sprint 03/Atividade 11/1.2/tarefa.py
from sys import maxsize

numero = 0
maior = 0
menor = maxsize
contadorNumeros = 0
NUMERO_DE_SAIDA = -1

while(True):
	contadorNumeros += 1
	
	#	Reinicia o valor do numero
	numero = NUMERO_DE_SAIDA - 1
	numero = int(input("Digite um numero: "))

	while(numero < NUMERO_DE_SAIDA):
		print("Valor invalido: numero digitado menor que 0.")
		numero = int(input("Digite um numero: "))

	if(numero == NUMERO_DE_SAIDA):
		break

	if(numero > maior):
		maior = numero
	if(numero < menor):
		menor = numero

if contadorNumeros == 1:
	print("Nenhum valor valido foi digitado!")
else:
	print("O menor numero natural digitado foi " + str(menor))
	print("O maior numero natural digitado foi " + str(maior))

File: Sprint 03/Atividade 11/1.2/Tarefa.java
import java.util.Scanner;
public class Tarefa {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		long maior = 0;
		long menor = Integer.MAX_VALUE;
		int contadorNumeros = 0;
		final int NUMERO_DE_SAIDA = -1;
		
		while(true) {
			contadorNumeros++;
			
			long numerosDigitados = 0;
			do{
				System.out.print("Digite o " + contadorNumeros + "-esimo numero natural: ");
				numerosDigitados = scan.nextLong();
			if(numerosDigitados <= -2){
			    
			    System.out.print("Valor invalido: numero digitado menor que 0 e diferente de -1.\n");
			}	
				
			}while(numerosDigitados < NUMERO_DE_SAIDA);
			
			if(numerosDigitados == NUMERO_DE_SAIDA) break;
	
			maior = (numerosDigitados > maior) ? numerosDigitados : maior;
			menor = (numerosDigitados < menor) ? numerosDigitados : menor;
		}

		if(contadorNumeros == 1){
			System.out.printf("Nenhum numero valido foi digitado!\n");
		}else {
			System.out.printf("O menor numero natural digitado foi " + menor + "!\n");
			System.out.printf("O maior numero natural digitado foi " + maior + "!\n");
		}

		scan.close();
	}
}